state/postgresql/postgresql_query.go

```diff
@@ -140,8 +139,8 @@ func (q *Query) Finalize(filters string, qq *query.Query) error {
 	return nil
 }
 
-func (q *Query) execute(ctx context.Context, logger logger.Logger, db *sql.DB) ([]state.QueryItem, string, error) {
-	rows, err := db.QueryContext(ctx, q.query, q.params...)
+func (q *Query) execute(ctx context.Context, logger logger.Logger, db dbquerier) ([]state.QueryItem, string, error) {
+	rows, err := db.Query(ctx, q.query, q.params...)
```

Comment:
should `defer rows.Close()` be called before error check? 
```
The returned Rows must be closed before the connection can be used again. It is safe to attempt to read from the returned Rows even if an error is returned. The error will be the available in rows.Err() after rows are closed. It is allowed to ignore the error returned from Query and handle it in Rows.
```
See https://pkg.go.dev/github.com/jackc/pgx/v5#Conn.Query

Reply:
Looks like the Close() call should be done after the error check. See example: https://pkg.go.dev/github.com/jackc/pgx/v5#example-Conn.Query